beacon-chain/core/validators/validator.go

```diff
@@ -634,3 +634,99 @@ func isActiveValidator(validator *pb.ValidatorRecord) bool {
 	return validator.Status == pb.ValidatorRecord_ACTIVE_PENDING_EXIT ||
 		validator.Status == pb.ValidatorRecord_ACTIVE
 }
+
+// activateValidator takes in validator index and activates
+// validator from pending activation status to active status.
+func activateValidator(state *pb.BeaconState, index uint32) (*pb.BeaconState, error) {
+	validator := state.ValidatorRegistry[index]
+	if validator.Status != pb.ValidatorRecord_PENDING_ACTIVATION {
+		return nil, fmt.Errorf("could not activte validator %d, status is %d",
+			index, validator.Status)
+	}
+	validator.Status = pb.ValidatorRecord_ACTIVE
+	validator.LatestStatusChangeSlot = state.Slot
+	newChainTip, err := NewRegistryDeltaChainTip(
+		pb.ValidatorRegistryDeltaBlock_ACTIVATION,
+		index,
+		validator.Pubkey,
+		state.ValidatorRegistryDeltaChainTipHash32,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("could not get new chain tip %v", err)
+	}
+	state.ValidatorRegistryDeltaChainTipHash32 = newChainTip[:]
+	state.ValidatorRegistry[index] = validator
+	return state, nil
+}
+
+// initiateValidatorExit takes in validator index and exits
+// validator from active status to active pending exit status.
+func initiateValidatorExit(state *pb.BeaconState, index uint32) (*pb.BeaconState, error) {
+	validator := state.ValidatorRegistry[index]
+	if validator.Status != pb.ValidatorRecord_ACTIVE {
+		return nil, fmt.Errorf("could not exit validator %d, status is %d",
```

Comment:
Should we print the expectation in the error message? i.e. expected validator to be active but was %s? Something like that

Reply:
sure, ill change it to that format